src/Parts.elm

```diff
@@ -87,268 +135,222 @@ type alias Update' model msg =
-  )
-  
-
--- EMBEDDING MESSAGES
-
+ -> Index
+ -> Accessors model container
+accessors collection model0 idx =
+  let
+    get container =
+      Dict.get idx (collection.get container) |> Maybe.withDefault model0
 
-{-| Similar to how embeddings enable collecting models of different type
-in a single model c, we collect messages in a single "master
-message" type. Messages exist exclusively to be dispatched by a corresponding
-`update` function; we can avoid distinguishing between different types of 
-messages by dispatching not the `Msg` itself, but a partially applied update
-function `update msg`. 
+    set container model = 
+      collection.set container (Dict.insert idx model (collection.get container))
+  in
+    { empty = model0
+    , get = get
+    , set = set
+    , map = \f c -> get c |> f |> set c
+    , reset = \c -> collection.get c 
+                 |> Dict.remove idx 
+                 |> collection.set c
+    }
 
-It's instructive to compare `Msg` to the type of `update` partially applied to 
-an actual carried message `m`:
 
-    update : m -> c -> (c, Cmd m)
-    (update m) : c -> (c, Cmd m)
--}
-type Msg c = 
-  Msg (c -> (Maybe c, Cmd (Msg c)))
+{-| Generate accessors for a component, that has only one instance.
 
+    instance = 
+      accessors1 .myField (\x y -> { y | myField = x }) init
 
-{-| Generic update function for `Msg`. 
 -}
-update : (Msg c -> m) -> Msg c -> c -> ( c, Cmd m )  
-update fwd (Msg f) c = 
-  f c 
-    |> map1st (Maybe.withDefault c)
-    |> map2nd (Cmd.map fwd)
-  
-
-{-| Generic explict no-op update function for `Msg`. 
--}
-update' : (Msg c -> m) -> Msg c -> c -> ( Maybe c, Cmd m )  
-update' fwd (Msg f) c = 
-  f c 
-    |> map2nd (Cmd.map fwd)
-  
-
--- PARTS
+accessors1
+  : (container -> model)
+ -> (container -> model -> container)
+ -> model
+ -> Accessors model container
+accessors1 get set model0 =
+  { empty = model0
+  , get = get
+  , set = set
+  , map = \f c -> get c |> f |> set c
+  , reset = flip set model0
+  }
 
 
-{-| Partially apply an `Update` function to a `msg`, producing a generic Msg.
--}
-pack : (Update c msg) -> msg -> Msg c 
-pack upd msg = 
-  Msg (\c -> 
-    upd msg c 
-      |> map1st Just
-      |> map2nd (Cmd.map (pack upd)))
+{-| Create a viewable component.
 
+    For your component:
+    render =
+      create view find
 
-{-| Partially apply an explicit no-op `Update'` function to a `msg`, producing
-a generic Msg.
+    In your main view:
+    Component.render ComponentMsg [0] model
 -}
-pack' : (a -> b -> ( Maybe b, Cmd a )) -> a -> Msg b
-pack' upd msg = 
-  Msg (\c -> 
-    upd msg c 
-      |> map2nd (Cmd.map (pack' upd)))
+create
+   : View model innerMsg 
+  -> (Index -> Accessors model container)
+  -> ((innerMsg, Index) -> outerMsg)
+  -> Index
+  -> container
+  -> Html outerMsg
+create view access wrapper idx container = 
+  App.map (\msg -> wrapper (msg, idx)) (view ((access idx).get container))
```

Comment:
I see why you restricted the type `View`. Here's an idea: Push the `App.map` into the `view` function (i.e., we require of `view` that it knows itself how to lift its messages, perhaps by applying `App.map` internally). The type of View is now

    type alias View = (Msg -> m) -> Model -> a

Then `create` should look something like

    create view access wrapper idx container = 
      view (\msg -> wrapper (msg, idx)) ((access idx).get) container

Obviously, this means we now expect `view` functions to have non-standard shape; this is not too bad since we're using a known trick (this was up on elm-discuss).